src/packet.rs

```diff
@@ -199,11 +199,16 @@ impl Packet {
                 )));
             }
         };
+        let tmst = if let Some(rx2) = &self.rx2_window {
```

Comment:
i'm not sure if this precedence logic is correct since the `to_pull_resp` method above takes an additional argument to direct it whether or not to use the `rx2_window` field if present.

Reply:
Yeah that's not right. the rx2 window is only used if rx1 fails. 

1. Have this function take a timestamp to declare the report at. Then have the code handling the BeaconResp use the timestamp it gets back
2. Make the timestamp use now to get the current time for the witness report

I think the first option seems closest to right?

Reply:
@Vagabond correct me if i misunderstood but i thought this field was intended to track the "radio clock" which wouldn't be the same thing as the system clock and i'm guessing rust would be deriving `now()` from the system clock? so where should the code handling the beacon resp get such a timestamp? wouldn't it get the timestamp from the beacon which would be what i have it doing right now and just switch the precedence order so it uses `self.timestamp` first and if that's `None` use the rx2 version?

Reply:
tmst is used a few different ways, rx2 only matters on scheduling a transmit, but in this case we've attached the time of the *successful* transmit to the txpk_ack, so I don't understand what were doing with rx2 here

Reply:
Resolve by removing for now and just using self.timestamp